src/components/app/DogFinder.js

```diff
@@ -1,58 +1,111 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { Flex, Box } from '@rebass/grid';
-import Slider from '../core/Slider';
+import Select from '../core/Select';
 import Widget from '../core/Widget';
-import Dogs from './Dogs';
+import VisuallyHidden from '../core/VisuallyHidden';
+import LoadingContainer from '../core/LoadingContainer';
+import CardList from './CardList';
+import DogCard from './DogCard';
+import { STATUSES } from '../../consts';
 
-const SliderColumn = ({ children }) => (
-	<Box width={[1, 1 / 3]} px={3} py={[3, 0]}>
-		{children}
-	</Box>
+const FilterSelect = ({ name, options, value, disabled, onChange }) => (
+	<Select
+		fullWidth
+		disabled={disabled || options.length <= 1}
+		value={value}
+		onChange={event => {
+			const { value } = event.target;
+			const number = Number(value);
+			onChange(name, String(number) === value ? number : value);
+		}}
+	>
+		{options.map(option => (
+			<option key={option.value} value={option.value}>
+				{option.label}
+			</option>
+		))}
+	</Select>
 );
 
-const DogFinder = ({ filters, dogs, onFilterUpdate }) => (
+const DogFinder = ({
+	status,
+	filterOptions,
+	filters,
+	dogs,
+	onFilterUpdate,
+}) => (
 	<React.Fragment>
+		<VisuallyHidden>
+			<h3>Filters</h3>
+		</VisuallyHidden>
 		<Flex mt={3} mb={5} mx={-3} flexWrap="wrap">
-			<SliderColumn>
-				<Widget label="Size">
-					<Slider
-						value={filters.size}
-						onChange={x => onFilterUpdate('size', x)}
-					/>
-				</Widget>
-			</SliderColumn>
-			<SliderColumn>
-				<Widget label="Energy">
-					<Slider
-						value={filters.energy}
-						onChange={x => onFilterUpdate('energy', x)}
-					/>
-				</Widget>
-			</SliderColumn>
-			<SliderColumn>
-				<Widget label="Barkness">
-					<Slider
-						value={filters.barkness}
-						onChange={x => onFilterUpdate('barkness', x)}
-					/>
-				</Widget>
-			</SliderColumn>
+			{filterOptions.map(({ name, label, options }) => (
+				<Box key={name} width={[1, 1 / 3]} px={3} py={[3, 0]}>
+					<Widget label={label}>
+						<FilterSelect
+							name={name}
+							options={options}
+							value={filters[name]}
+							disabled={status === STATUSES.LOADING}
+							onChange={onFilterUpdate}
+						/>
+					</Widget>
+				</Box>
+			))}
 		</Flex>
-		<Dogs dogs={dogs} />
+		<VisuallyHidden>
+			<h3>
+				Search results{' '}
+				<span aria-live="polite" role="status">
```

Comment:
Same...

Reply:
What kind of a component you'd use here?

Reply:
I'd create a special component if that's a pattern or would use a `Text` with appropriate props

Reply:
Again: it's not text because it's invisible. And I don't think ARIA is a pattern, I'd like to use it on any element or component if that makes sense, and if it's impossible to hardcode inside a component.